packages/api/src/tx/modules/ecocredit/v1/create_batch_amino.ts

```diff
@@ -60,20 +60,21 @@ export function createBatchConverter(): AminoConverter {
         project_id: projectId,
         issuance: issuance.map(i => {
           return {
-            $type: BatchIssuance.$type,
-            recipient: i.recipient,
-            tradable_amount: i.tradableAmount,
-            retired_amount: i.retiredAmount,
-            retirement_jurisdiction: i.retirementJurisdiction,
-          };
+            recipient: i.recipient || undefined,
+            tradable_amount: i.tradableAmount || undefined,
+            retired_amount: i.retiredAmount || undefined,
+            retirement_jurisdiction: i.retirementJurisdiction || undefined,
+          }
         }),
         metadata,
-        start_date: startDate,
-        end_date: endDate,
+        start_date: AminoDate(startDate),
+        end_date: AminoDate(endDate),
```

Comment:
```suggestion
        start_date: AminoDate(startDate) || undefined,
        end_date: AminoDate(endDate) || undefined,
```

^ so I think we would add stuff like this wherever we have optional fields. `AminoDate()` would render a null proto date as an empty string, and then we'd convert it to `undefined` so it gets omitted when the amino JSON gets serialized.